src/clients/TorrentClient.ts

```diff
@@ -1,25 +1,51 @@
 import { Metafile } from "../parseTorrent.js";
-import { DecisionAnyMatch, InjectionResult } from "../constants.js";
+import {
+	Decision,
+	DecisionAnyMatch,
+	InjectionResult,
+	VIDEO_DISC_EXTENSIONS,
+} from "../constants.js";
 import { getRuntimeConfig } from "../runtimeConfig.js";
 import { Searchee, SearcheeWithInfoHash } from "../searchee.js";
 import QBittorrent from "./QBittorrent.js";
 import RTorrent from "./RTorrent.js";
 import Transmission from "./Transmission.js";
 import Deluge from "./Deluge.js";
 import { Result } from "../Result.js";
+import { Label } from "../logger.js";
+import { hasExt } from "../utils.js";
 
-let activeClient: TorrentClient;
+let activeClient: TorrentClient | null = null;
+
+export type TorrentClientType =
+	| Label.QBITTORRENT
+	| Label.RTORRENT
+	| Label.TRANSMISSION
+	| Label.DELUGE;
+
+export interface GenericTorrentInfo {
+	infoHash: string;
+	category: string;
+	tags: string[];
+	trackers?: string[][];
+}
 
 export interface TorrentClient {
+	type: TorrentClientType;
```

Comment:
what is this for? can we keep it separate from logging labels?

Reply:
> Type allows handling qbit specifically since we read the data from fastresume usually.

I see. I don't really want to introduce a dependency on fastresume files which are a private API for qbittorrent (and being deprecated!).

 If we already support this via client calls what is the point of reading the fastresume files?

Reply:
I used labels as it's one less thing to import and use. And it will always exist for any future clients.

We have to read the .fastresume files to get the tracker information, so we might as well read category and tags at the same time. The current api call only returns the active tracker (nothing if paused) when getting all torrents. To get all trackers regardless of state would require a separate api call for each torrent. Moving to the pure api method for qbit will need to cache that value between `cross-seed` restarts since they won't change (user can add tracker but we don't support modifying torrents anyways).